assets/js/components/ResetButton.test.js

```diff
@@ -0,0 +1,109 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { render, createTestRegistry, fireEvent, act } from '../../../tests/js/test-utils';
+import { STORE_NAME as CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import ResetButton from './ResetButton';
+import { subscribeUntil } from '../../../tests/js/utils';
+
+describe( 'ResetButton', () => {
+	let registry;
+	const adminURL = 'http://example.com/';
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		registry.dispatch( CORE_SITE ).receiveSiteInfo( { adminURL } );
+	} );
+
+	it( 'should render "Reset Site Kit" as the default child', () => {
+		const { container } = render( <ResetButton />, { registry } );
+		expect( container ).toHaveTextContent( 'Reset Site Kit' );
+	} );
+
+	it( 'should render passed children inside of the button', () => {
+		const { container } = render( <ResetButton>Test Value</ResetButton>, { registry } );
+		expect( container ).toHaveTextContent( 'Test Value' );
+		expect( document.querySelector( '.mdc-dialog--open' ) ).toBeNull();
+	} );
+
+	describe( 'after click', () => {
+		let container;
+		beforeEach( () => {
+			( { container } = render( <ResetButton />, { registry } ) );
+			fireEvent.click( container.querySelector( '.googlesitekit-reset-button' ) );
+		} );
+
+		it( 'should open the dialog', async () => {
+			expect( document.querySelector( '.mdc-dialog--open' ) ).toBeTruthy();
+		} );
+
+		it( 'should show reset and cancel buttons', async () => {
+			expect( document.querySelector( '.mdc-dialog--open .mdc-dialog__cancel-button' ) ).toBeTruthy();
+			expect( document.querySelector( '.mdc-dialog--open .mdc-button--danger' ) ).toBeTruthy();
+		} );
+
+		it( 'should close the modal on clicking cancel', async () => {
+			fireEvent.click( document.querySelector( '.mdc-dialog--open .mdc-dialog__cancel-button' ) );
+			expect( document.querySelector( '.mdc-dialog--open' ) ).toBeNull();
+		} );
+
+		it( 'should close the modal on pressing escape key', async () => {
+			fireEvent.keyUp( global, { keyCode: 27 } );
+			expect( document.querySelector( '.mdc-dialog--open' ) ).toBeNull();
+		} );
+
+		it( 'should close the modal on clicking Reset and ', async () => {
+			const response = true;
+			fetchMock.postOnce(
+				/^\/google-site-kit\/v1\/core\/site\/data\/reset/,
+				{ body: JSON.stringify( response ), status: 200 },
+			);
+
+			delete global.location;
```

Comment:
Why do we delete the location here as part of the test?

Reply:
@kostyalmm – you resolved the question here without a reply; would you please clarify?

Reply:
@aaemnnosttv I had removed this statement earlier and the code worked without it. But with the latest(a couple of days ago) requested changes, I had to reintroduce this. I had forgotten about this comment from earlier. I'll take care of such cases moving forward.

Reply:
Why I reintroduced it:
For some reason if I just reassign `global.location` and set its `assign` property to a jest mock function, it does not work properly.
I tried doing it without the `delete`, but the mock function was not being called at all.
After researching online, I noticed that all the people are deleting the `global.location` object for the same use case. When I deleted it, the test worked fine.
I am not aware of the reason why this happens, maybe some reference issues, but deleting the location object solves it.

Reply:
Thanks for the context @kostyalmm – I took a look into this and found that it has to do with restrictions introduced in a recent version of jsdom.

You are correct that we can redefine the `location` global, we just can't change it. I found we can keep everything else like so:

```js
	Object.defineProperty( global, 'location', {
		writable: true,
		value: {
			...global.location,
			assign: jest.fn(),
		},
	} );
```

**Let's keep the location mocking in this test for now** and we can come back to this with a follow-up issue since this should probably be baked into the test setup in a top-level `beforeAll`.

One thing we should do here is backup the original "real" `location` and restore it at the end so that other tests aren't affected by this since it is not reset between tests as it is.